osCam/userconfig/views.py

```diff
@@ -1,111 +1,231 @@
-            # print(key, meta[key] )    # print this to see all Network Information supplied by Django and network packet. If we want more we need send out ARP or get from user directly. 
-    # Network(request.get_host())
-    return JsonResponse({'response': 'SUCCESS', 'type':'GET', 'Client IP Address': network_info['REMOTE_ADDR'], 'PAGE': page}, safe=True)
-
-
-
-def editConf(request, id):
-    '''
-        View Handles response To [EDIT] a Configuration of Type::{RaspberryPi, Camera, Network}
-        
-        @param  request: Request 
-                id: Integer
-    '''
-    page = "Edit Config"
-    return JsonResponse({'response': 'SUCCESS', 'type':'GET', 'PAGE': page}, safe=True)
-
-
-
-
-def removeConf(request, id):
-    '''
-        View Handles response To [REMOVE] a Configuration of Type::{RaspberryPi, Camera, Network}
-        @param  request: Request 
-                id: Integer
-    '''
-    page = "Remove Config"
-    return JsonResponse({'response': 'SUCCESS', 'type':'GET', 'PAGE:': page}, safe=True)
\ No newline at end of file
+    page_data = {
+        'cameras': Camera.objects.all(), 
+        'views': CameraView.objects.all(), 
+        'network': Network.objects.get(id=1),
+        'storages': Storage.objects.all(),
+        'raspberrypis': RaspberryPi.objects.all(),
+        }
+
+    # save camera settings 
+    if(request.method == 'POST'):
+        if("add_camera_config" in request.POST):
+            add_form = CameraEntryForm(request.POST)
+            if(add_form.is_valid()):
+                deviceName = add_form.cleaned_data['deviceName']
+                ipAddress = add_form.cleaned_data['ipAddress']
+                port = add_form.cleaned_data['port']
+
+                this_cam_instance = Camera.objects.get(id=1)
+                this_cam_instance.deviceName = deviceName
+                this_cam_instance.ipAddress = ipAddress
+                this_cam_instance.port = port
+                this_cam_instance.save()
+
+            return redirect('user/settings.html')
+
+        # save view settings
+        elif("add_view_config" in request.POST):
+            add_form = CameraViewForm(request.POST)
+            if(add_form.is_valid()):
+                showMotionBoxes = add_form.cleaned_data['showMotionBoxes']
+                showContours = add_form.cleaned_data['showContours']
+                showText = add_form.cleaned_data['showText']
+                text = add_form.cleaned_data['text']
+                contrast = add_form.cleaned_data['contrast']
+                brightness = add_form.cleaned_data['brightness']
+                recording = add_form.cleaned_data['recording']
+                fps = add_form.cleaned_data['fps']
+                invert = add_form.cleaned_data['invert']
+                mirror = add_form.cleaned_data['mirror']
+
+                this_view_instance = CameraView.objects.get(id=1)
```

Comment:
I think "this_view_instance" could be better named

Reply:
lol sure